File: SteamTokenGenerator.cs
using Newtonsoft.Json.Linq;
using System;
using System.IO;
using System.Net;

namespace SteamTokenGenerator
{
	class SteamTokenGenerator
	{
		static readonly string MainPath = Environment.GetFolderPath( Environment.SpecialFolder.MyDocuments ) + @"\SteamTokenGenerator";

		static string GetKey()
		{
			string readkey = File.ReadAllText( MainPath + @"\key.txt" );
			Console.WriteLine( "\nReading user key..." );
			return readkey;
		}

		static string GetJSON()
		{
			string URL = "https://api.steampowered.com/IGameServersService/GetAccountList/v1/?key=" + GetKey();
			string response;
			Console.WriteLine( "\nGrabbing JSON..." );
			WebRequest tokenRequest = WebRequest.Create( URL );
			WebResponse getToken = tokenRequest.GetResponse();
			using ( Stream dataStream = getToken.GetResponseStream() )
			{
				StreamReader reader = new StreamReader( dataStream );
				response = reader.ReadToEnd();
			}
			getToken.Close();
			Console.WriteLine( "\nReading JSON..." );
			return response;
		}

		static void WriteKey( string token )
		{
			Console.WriteLine( "\nWriting tokens to file..." );
			File.WriteAllText( MainPath + @"\token.txt", token );
		}

		static void CheckToken()
		{
			Console.WriteLine( "\nParsing JSON..." );
			dynamic jsonlist = JObject.Parse( GetJSON() );
			dynamic servers = jsonlist.response.servers;
			bool FoundUsedTokens = false;
			Console.WriteLine( "\nSearching for old tokens..." );
			int count = 0;
			foreach ( dynamic token in servers )
			{
				if ( token.is_expired == true )
				{
					string deleteurl = "https://api.steampowered.com/IGameServersService/DeleteAccount/v1/?key=" + GetKey() + "&steamid=" + token.steamid;
					WebRequest deleterequest = WebRequest.Create( deleteurl );
					deleterequest.Method = "POST";
					deleterequest.ContentType = "application/x-www-form-urlencoded";
					deleterequest.GetResponse();
					FoundUsedTokens = true;

					Console.WriteLine( "\nCreating new token..." );
					string createurl = "https://api.steampowered.com/IGameServersService/CreateAccount/v1/?key=" + GetKey() + "&appid=4000&memo=AutoToken" + count;
					WebRequest createrequest = WebRequest.Create( createurl );
					createrequest.Method = "POST";
					createrequest.ContentType = "application/x-www-form-urlencoded";
					
					dynamic newjson = JObject.Parse( GetJSON() );
					dynamic newservers = newjson.response.servers;
					string tokenstr = servers[0].login_token;
					WriteKey( tokenstr );
				}
				count++;
			}
			Console.WriteLine( FoundUsedTokens ? "\nRefreshing old tokens..." : "\nNo old tokens found." );
		}

		static void Main( string[] args )
		{
			Console.WriteLine( "\nInitializing..." );
			CheckToken();
			Console.WriteLine( "\nProcess finished." );
		}
	}
}

File: SteamTokenGenerator.py
import json
import os
import requests
import sys

MainPath = os.path.expanduser( "~/Documents/SteamTokens" )

def GetJSON():
	print( "\nGrabbing JSON..." )
	jsonurl = requests.get( f"https://api.steampowered.com/IGameServersService/GetAccountList/v1/?key={Key}" )
	print( "\nReading JSON..." )
	return json.loads( jsonurl.text )

def WriteKey( name, token ):
	print( "\nWriting tokens to file..." )
	f = open( MainPath + f"/{name}.txt", "w+" )
	f.write( token )
	f.close()

def CheckToken():
	jsonfile = GetJSON()
	servers = jsonfile["response"]["servers"] if "servers" in jsonfile["response"] else None
	foundusedtokens = False
	print( "Searching for old tokens..." )
	count = 0

	if servers is None:
		print( f"No tokens found. Creating a new one for appid {AppID}..." )
		memo = f"AutoToken{AppID}{count}"
		requests.post( f"https://api.steampowered.com/IGameServersService/CreateAccount/v1/?key={Key}&appid={AppID}&memo={memo}" )
		newjson = GetJSON()
		newservers = newjson["response"]["servers"]
		tokenstr = newservers[0]["login_token"]
		WriteKey( memo, tokenstr )
		return

	for token in servers:
		if token["is_expired"]:
			requests.post( f"https://api.steampowered.com/IGameServersService/DeleteAccount/v1/?key={Key}&steamid={token['steamid']}" )
			foundusedtokens = True
			print( "\nCreating new token..." )
			memo = f"AutoToken{AppID}{count}"
			requests.post( f"https://api.steampowered.com/IGameServersService/CreateAccount/v1/?key={Key}&appid={AppID}&memo={memo}" )
			newjson = GetJSON()
			newservers = newjson["response"]["servers"]
			tokenstr = newservers[count]["login_token"]
			WriteKey( memo, tokenstr )
		count += 1
	print( foundusedtokens and "\nRefreshing old tokens..." or "\nNo old tokens found." )

if __name__ == "__main__":
	if not os.path.exists( MainPath ):
		os.makedirs( MainPath )

	if len( sys.argv ) < 2:
		print( "Arguments not provided. Aborting." )
		exit()
	
	global Key, AppID
	AppID = sys.argv[1]
	Key = sys.argv[2]
	print( "Initializing..." )
	CheckToken()
	print( "Process finished." )
